core/src/test/java/org/apache/iceberg/TestOverwrite.java

```diff
@@ -182,16 +182,16 @@ public void testOverwriteFailsDelete() {
   @Test
   public void testOverwriteWithAppendOutsideOfDelete() {
     TableMetadata base = TestTables.readMetadata(TABLE_NAME);
-    long baseId =
-        latestSnapshot(base, branch) == null ? -1 : latestSnapshot(base, branch).snapshotId();
+    Snapshot latestSnapshot = latestSnapshot(base, branch);
```

Comment:
We don't need to handle in this PR but given #6651 we will end up using the SnapshotUtil api which is introduced. We can either update this PR with the proposed utility API or just later refactor to use the utility API. Since this is a test class, I'm okay either way.

Reply:
maybe we can just add the `SnapshotUtil` change in this PR? I don't think we can resolve the comments in that PR in time based on the current situation.